src/NuGet.Clients/PackageManagement.UI/Xamls/InfiniteScrollList.xaml.cs

```diff
@@ -182,21 +191,19 @@ private void LoadItems(PackageItemListViewModel selectedPackageItem)
             token.ThrowIfCancellationRequested();
 
             var loadedItems = await LoadNextPageAsync(currentLoader, token);
-
-            token.ThrowIfCancellationRequested();
-
-            // multiple loads may occur at the same time
-            if (currentLoader == _loader)
-            {
-                UpdatePackageList(loadedItems.ToList(), refresh: false);
-            }
-
             token.ThrowIfCancellationRequested();
 
             await NuGetUIThreadHelper.JoinableTaskFactory.RunAsync(async () =>
             {
                 await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
-                _loadingStatusBar.ItemsLoaded = currentLoader.State.ItemsCount;
+
+               // multiple loads may occur at the same time
```

Comment:
I don't understand what this comment means. Can you make it clearer?

Also is there a race condition here?